### ISSUE TITLE
[harvester global flag] toggle harvester node driver with the harvester global flag

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#34886]
**Is your feature request related to a problem? Please describe.**
* Currently harvester node driver is disabled by default. 

**Describe the solution you'd like**
I think we should enable this when harvester global flag is enabled (which is by default).

**Comment by yasker** [comment]
I'd like to enable Harvester node driver by default as well, as long as there is no objection from @cloudnautique .

Also, I don't think we need to do sync between the two flags at this moment.

**Comment by cloudnautique** [comment]
It should be enabled by default. If the user disables the flag, then disable the driver. The admin should still be able to enable the driver manually in the driver management page.

**Comment by yasker** [comment]
Move to v2.6.2 since it's not critical enough for a last-minute squeeze for v.2.6.1.

**Comment by yasker** [comment]
Update from @cloudnautique 

```
I like the idea of enabling both with the harvester flag. It gives an easy all in experience. For toggling it off, we can keep them disconnected.
```

So:
1. If the feature flag was turned on, we enable the Harvester node driver automatically (if it's not enabled).
2. If the feature flag was turned off, nothing will change to the Harvester node driver.
3. Enable/disable the Harvester node driver will not affect the state of the feature flag.

**Comment by TachunLin** [comment]
Verified fixed on Rancher v2.6.3-rc1 with builtin node driver 
Pass the following test scenario. Close this issue and [harvester/harvester#1465](https://github.com/harvester/harvester/issues/1465) accordingly

Test in Rancher v2.6.3-rc1 under airgapped environment


- [x] Harvester feature flag will be enabled by default and turned on harvester node driver accordingly

- [x] If the feature flag was turned off, nothing will change to the Harvester node driver.

- [x] Enable/disable the Harvester node driver will not affect the state of the feature flag.

- [x] If the feature flag was turned on, we enable the Harvester node driver automatically
